File: mark-track/backend/models/notification.py
from typing import Optional
from pydantic import BaseModel

class MarkNotification(BaseModel):
    student_id: str
    teacher_id: str
    subject_id: str
    value: float
    description: Optional[str] = None
    is_read: bool = False

class AbsenceNotification(BaseModel):
    student_id: str
    teacher_id: str
    subject_id: str
    is_motivated: bool
    date: str
    description: Optional[str] = None
    is_read: bool = False

class ClassAsignmentNotification(BaseModel):
    pass
    

File: mark-track/src/types/notification.ts
export interface NotificationBase {
    student_id: string;
    teacher_id: string;
    subject_id: string;
}

export interface MarkNotification extends NotificationBase {
    id: string;
    value: number;
    description: string;
    date: string;
}

export interface AbsenceNotification extends NotificationBase {
    id: string;
    is_motivated: boolean;
    description: string;
    date: string;
}